### PR TITLE
[SR-4816] fix external hive scan: type mismatch and logical empty file

### PR DESCRIPTION
**Pull request by dirtysalt**
For external hive scan, we report errors if we find column tye mismatch.

And we may encounter a logical empty file during scan: file size != 0, but returns EOF immediately.  We can not return once we encounter a logical empty file, because there are many other files be scanned in other threads.

### CODE REVIEW

File: be/src/exec/vectorized/hdfs_scan_node.cpp [hunk 1]
```diff
@@ -261,9 +261,6 @@ int HdfsScanNode::_compute_priority(int32_t num_submitted_tasks) {
 
 void HdfsScanNode::_scanner_thread(HdfsScanner* scanner) {
     Status status = scanner->open(_runtime_state);
-    if (!status.ok()) {
```
Comment (dirtysalt) [inline be/src/exec/vectorized/hdfs_scan_node.cpp:264]: We may encounter a logical empty file, but we can not set EOF right here. There are many other files be scanned.

File: be/src/exec/vectorized/orc_scanner_adapter.cpp [hunk 2]
```diff
@@ -1196,6 +1196,15 @@ Status OrcScannerAdapter::_init_cast_exprs() {
             _cast_exprs[column_pos] = slot;
             continue;
         }
+        // we don't suppot cast column in query external hive table case.
```
Comment (wyb) [inline be/src/exec/vectorized/orc_scanner_adapter.cpp:1199]: support

**Comment by kangkaisen** [comment]
This pr fix two bug?

File: be/src/exec/vectorized/orc_scanner_adapter.cpp [hunk 3]
```diff
@@ -937,8 +937,8 @@ OrcScannerAdapter::OrcScannerAdapter(const std::vector<SlotDescriptor*>& src_slo
           _tzinfo(cctz::utc_time_zone()),
           _tzoffset_in_seconds(0),
           _drop_nanoseconds_in_datetime(false),
-          _broker_load_mode(false),
-          _strict_mode(false),
+          _broker_load_mode(true),
+          _strict_mode(true),
```
Comment (Seaven) [inline be/src/exec/vectorized/orc_scanner_adapter.cpp:941]: Why modify default _strict_mode? It's will effect load.

Comment (dirtysalt) [reply be/src/exec/vectorized/orc_scanner_adapter.cpp:941]: It's just a default value.  The user code will set it properly when to use it.

in orc_scanner.cpp
```
    _orc_adapter = std::make_unique<OrcScannerAdapter>(_orc_slot_descriptors);
    _orc_adapter->set_broker_load_mode(_strict_mode);
```

and in hdfs_scanner_orc.cpp

```
    _orc_adapter = std::make_unique<OrcScannerAdapter>(_src_slot_descriptors);
    _orc_row_reader_filter =
            std::make_shared<OrcRowReaderFilter>(_scanner_params, _file_read_param, _orc_adapter.get());
    _orc_adapter->disable_broker_load_mode();
```

**Comment by dirtysalt** [comment]
> This pr fix two bug?

Yes, two bugs in a single case.

**Review by Seaven — APPROVED** [review]

File: be/src/exec/vectorized/orc_scanner_adapter.cpp [hunk 4]
```diff
@@ -1196,6 +1196,15 @@ Status OrcScannerAdapter::_init_cast_exprs() {
             _cast_exprs[column_pos] = slot;
             continue;
         }
+        // we don't support implicit cast column in query external hive table case.
+        // if we query external table, we heavily rely on type match to do optimization.
+        // For example, if we assume column A is a integer column, but it's stored as string in orc file
+        // then min/max of A is almost unusable. Think that there are values [10, 11, 10000, 100001]
+        // min/max will be "10" and "11", and we expect min/max is 10/100001
+        if (!_broker_load_mode) {
```
Comment (kangkaisen) [inline be/src/exec/vectorized/orc_scanner_adapter.cpp:1204]: The mode means multi option，but we only have is load or is query. So I think the better name is `_is_broker_load`

**Review by kangkaisen — APPROVED** [review]

File: be/src/exec/vectorized/orc_scanner_adapter.cpp (hunk 4 above)
Comment (dirtysalt) [reply be/src/exec/vectorized/orc_scanner_adapter.cpp:1204]: Yes, `is_broker_load` is a better name.

**Pull request merged**